libs/src/api/serviceProvider.js

```diff
@@ -82,35 +93,46 @@ class ServiceProvider extends Base {
       creatorNodes = creatorNodes.filter(node => !blacklist.has(node.endpoint))
     }
 
-    // Filter to healthy nodes
-    creatorNodes = (await Promise.all(
+    // Get sync status for all nodes and filter out unhealthy (unresponsive) nodes.
+    const healthyCreatorNodesSyncStatus = {}
+    await Promise.all(
       creatorNodes.map(async node => {
         try {
-          const { isBehind, isConfigured } = await this.creatorNode.getSyncStatus(node.endpoint)
-          return isConfigured && isBehind ? false : node.endpoint
+          const syncStatus = await this.creatorNode.getSyncStatus(node.endpoint)
+          healthyCreatorNodesSyncStatus[node.endpoint] = syncStatus
         } catch (e) {
-          return false
+          // Ignore unhealthy nodes.
+          console.error(`get sync status failed ${node.endpoint}`)
         }
       })
-    ))
-      .filter(Boolean)
+    )
+    const healthyCreatorNodes = Object.keys(healthyCreatorNodesSyncStatus)
 
-    // Time requests and autoselect nodes
+    // Time requests and autoselect nodes.
     const timings = await Utils.timeRequests(
-      creatorNodes.map(node => ({
+      healthyCreatorNodes.map(node => ({
         id: node,
         url: `${node}/version`
       }))
     )
 
+    // Nodes version info object.
     let services = {}
     timings.forEach(timing => {
       services[timing.request.id] = timing.response.data
     })
-    // Primary: select the lowest-latency
-    const primary = timings[0] ? timings[0].request.id : null
 
-    // Secondaries: select randomly
+    let primaryOptions = timings
+    if (enforceSyncStatus) {
```

Comment:
this flag is there because this function is overloaded and also used to backfill a single secondary, along with selecting all 3 nodes at creator init.